### PR TITLE
organize readmes and examples

### PR DESCRIPTION
**Pull request by ritmun**
Organized readme and examples

docs go under docs/
examples go under examples/

Main readme doc leads to examples as follows: 

Main readme -> AWS specific readme -> examples and example readmes

### CODE REVIEW

File: examples/aws/firewall/firewallConfig.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package main
+package firewall
```
Comment (fahlmant) [inline examples/aws/firewall/firewallConfig.go:1]: Go programs require a main package. This change breaks the script

Comment (boranx) [reply examples/aws/firewall/firewallConfig.go:1]: I think in general we shouldn't propose changes in source code unless we're sure/executed in local and see it's working as expected

Comment (ritmun) [reply examples/aws/firewall/firewallConfig.go:1]: that was my IDE's brilliant suggestion :(

Comment (ritmun) [reply examples/aws/firewall/firewallConfig.go]: reverted and tested script locally

**Review by fahlmant — CHANGES_REQUESTED** [review]
The build is failing, looks like go.mod and go.sum are required

File: examples/aws/firewall/firewallConfig.go (hunk 1 above)
Comment (ritmun) [reply examples/aws/firewall/firewallConfig.go]: @boranx I agree with that. Also that every piece of code should have a test in the pipeline pre-merge to make sure nothing like that skips our attention.

**Comment by ritmun** [comment]
> The build is failing, looks like go.mod and go.sum are required

thanks, fixed

**Comment by boranx** [comment]
@ritmun could you rebase this so we can merge?

**Comment by ritmun** [comment]
> @ritmun could you rebase this so we can merge?

rebased

**Pull request merged**